src/python/pants/backend/jvm/tasks/jar_publish.py

```diff
@@ -868,8 +869,10 @@ def get_synthetic(lang, target):
     def exportable(tgt):
       return tgt in candidates and self._is_exported(tgt)
 
-    return OrderedSet(filter(exportable,
-                             reversed(sort_targets(filter(exportable, candidates)))))
+    return OrderedSet(target for target in
```

Comment:
This nested generator clause works, although is a little hard to read, as was the original. 

I can refactor if you want, although might be better to keep original structure.